packages/core/src/execution-engine/workflow-execute.ts

```diff
@@ -1182,151 +1183,305 @@ export class WorkflowExecute {
-					// If we would not be able to wait for it to close would it cause problems with "own" mode as the
-					// process would be killed directly after it and so the acknowledge would not have been finished yet.
-					closeFunction = triggerResponse.closeFunction;
-
-					// Manual testing of Trigger nodes creates an execution. If the execution is cancelled, `closeFunction` should be called to cleanup any open connections/consumers
-					abortSignal?.addEventListener('abort', closeFunction);
-				}
+	/**
+	 * Executes the given node based on its type and configuration
+	 *
+	 * @param workflow - The workflow containing the node
+	 * @param executionData - Data needed for node execution
+	 * @param runExecutionData - Overall execution state
+	 * @param runIndex - Index of current execution run
+	 * @param additionalData - Additional workflow execution data
+	 * @param mode - Workflow execution mode
+	 * @param abortSignal - Signal for canceling execution
+	 * @returns Promise resolving to node execution result
+	 */
+	async runNode(
+		workflow: Workflow,
+		executionData: IExecuteData,
+		runExecutionData: IRunExecutionData,
+		runIndex: number,
+		additionalData: IWorkflowExecuteAdditionalData,
+		mode: WorkflowExecuteMode,
+		abortSignal?: AbortSignal,
+	): Promise<IRunNodeResponse> {
+		const { node } = executionData;
+		let inputData = executionData.data;
 
-				if (triggerResponse.manualTriggerFunction !== undefined) {
-					// If a manual trigger function is defined call it and wait till it did run
-					await triggerResponse.manualTriggerFunction();
-				}
+		if (node.disabled === true) {
+			return this.handleDisabledNode(inputData);
+		}
 
-				const response = await triggerResponse.manualTriggerResponse!;
+		const nodeType = workflow.nodeTypes.getByNameAndVersion(node.type, node.typeVersion);
+		const isDeclarativeNode = nodeType.description.requestDefaults !== undefined;
+		const customOperation = this.getCustomOperation(node, nodeType);
 
-				if (response.length === 0) {
-					return { data: null, closeFunction };
-				}
+		const connectionInputData = this.prepareConnectionInputData(
+			workflow,
+			nodeType,
+			customOperation,
+			inputData,
+		);
 
-				return { data: response, closeFunction };
-			}
-			// For trigger nodes in any mode except "manual" do we simply pass the data through
-			return { data: inputData.main as INodeExecutionData[][] };
+		if (connectionInputData === null) {
+			return { data: undefined };
+		}
+
+		this.handlePreviousExecutionError(runExecutionData, node);
+
+		inputData = this.handleExecuteOnce(node, inputData);
+
+		if (nodeType.execute || customOperation) {
+			return await this.executeNode(
+				workflow,
+				node,
+				nodeType,
+				customOperation,
+				additionalData,
+				mode,
+				runExecutionData,
+				runIndex,
+				connectionInputData,
+				inputData,
+				executionData,
+				abortSignal,
+			);
+		} else if (nodeType.poll) {
```

Comment:
For a minimal readability gain in this nightmare file, since all these cases return, can we remove the unnecessary elses and spread out these cases?